File: examples/python/list-snapshots.py
#!/usr/bin/python3

from time import gmtime, asctime
from pwd import getpwuid
import dbus

bus = dbus.SystemBus()

snapper = dbus.Interface(bus.get_object('org.opensuse.Snapper', '/org/opensuse/Snapper'),
                         dbus_interface='org.opensuse.Snapper')


snapshots = snapper.ListSnapshots("root")

for snapshot in snapshots:
    print(snapshot[0], snapshot[1], snapshot[2], end='')

    if snapshot[3] != -1:
        print("", asctime(gmtime(snapshot[3])), end='')
    else:
        print("", "now", end='')

    print("", getpwuid(snapshot[4])[0], snapshot[5], snapshot[6], end='')
    for k, v in snapshot[7].items():
        print("", "%s=%s" % (k, v), end='')
    print()

File: examples/ruby/list-snapshots.rb
#!/usr/bin/ruby

require "dbus"

system_bus = DBus::SystemBus.instance

service = system_bus.service("org.opensuse.Snapper")

dbus_object = service.object("/org/opensuse/Snapper")

dbus_object.introspect
dbus_object.default_iface = "org.opensuse.Snapper"

snapshots = dbus_object.send("ListSnapshots", "root")[0]

snapshots.each do |snapshot|
  print snapshot, "\n"
end

